### PR TITLE
Added STAT_CHANGE_PRESERVE_MULTICHOOSE

### PR DESCRIPTION
**Pull request by tustin2121**
Added flag STAT_CHANGE_PRESERVE_MULTICHOOSE for ChangeStatBuffs to tell it not to write to MULTISTRING_CHOOSER.

## Description
- I found this necessary for a modification I did recently which added a variant Intimidate string. Without it, it would always choose the wrong string from a `printfromtable` command, because the stat change call was stomping on my previously set value to MULTISTRING_CHOOSER. There's really no need for it *always* to do the side effect of changing MULTISTRING_CHOOSER.
- This commit doesn't add uses of this new flag anywhere; it adds it for downstream users who want to add battle effects. Like myself.

Example:
```diff
BattleScript_IntimidateActivates::
	setbyte gBattlerTarget, 0
+	@ skip popup if holding intimidate item
+	jumpifbyte CMP_EQUAL, cMULTISTRING_CHOOSER, 1, BattleScript_IntimidateActivatesLoop
	call BattleScript_AbilityPopUp
BattleScript_IntimidateActivatesLoop:
	setstatchanger STAT_ATK, 1, TRUE
	trygetintimidatetarget BattleScript_IntimidateActivatesReturn
	jumpifstatus2 BS_TARGET, STATUS2_SUBSTITUTE, BattleScript_IntimidateActivatesLoopIncrement
	jumpifability BS_TARGET, ABILITY_CLEAR_BODY, BattleScript_IntimidatePrevented
	jumpifability BS_TARGET, ABILITY_HYPER_CUTTER, BattleScript_IntimidatePrevented
	jumpifability BS_TARGET, ABILITY_WHITE_SMOKE, BattleScript_IntimidatePrevented
.if B_UPDATED_INTIMIDATE >= GEN_8
	jumpifability BS_TARGET, ABILITY_INNER_FOCUS, BattleScript_IntimidatePrevented
	jumpifability BS_TARGET, ABILITY_SCRAPPY, BattleScript_IntimidatePrevented
	jumpifability BS_TARGET, ABILITY_OWN_TEMPO, BattleScript_IntimidatePrevented
	jumpifability BS_TARGET, ABILITY_OBLIVIOUS, BattleScript_IntimidatePrevented
.endif
-	statbuffchange STAT_CHANGE_NOT_PROTECT_AFFECTED | STAT_CHANGE_ALLOW_PTR, BattleScript_IntimidateActivatesLoopIncrement
+	statbuffchange STAT_CHANGE_NOT_PROTECT_AFFECTED | STAT_CHANGE_ALLOW_PTR | STAT_CHANGE_PRESERVE_MULTICHOOSE, BattleScript_IntimidateActivatesLoopIncrement
	jumpifbyte CMP_GREATER_THAN, cMULTISTRING_CHOOSER, 1, BattleScript_IntimidateActivatesLoopIncrement
	setgraphicalstatchangevalues
	playanimation BS_TARGET, B_ANIM_STATS_CHANGE, sB_ANIM_ARG1
-	printstring STRINGID_PKMNCUTSATTACKWITH
+	printfromtable gIntimidateStringIds
	waitmessage B_WAIT_TIME_LONG
	call BattleScript_TryAdrenalineOrb
```

## **Discord contact info**
Tustin2121#6219

### CODE REVIEW

File: src/battle_script_commands.c [hunk 1]
```diff
@@ -10623,11 +10624,14 @@ static u32 ChangeStatBuffs(s8 statValue, u32 statId, u32 flags, const u8 *BS_ptr
         gBattleMons[gActiveBattler].statStages[statId] = MIN_STAT_STAGE;
     if (gBattleMons[gActiveBattler].statStages[statId] > MAX_STAT_STAGE)
         gBattleMons[gActiveBattler].statStages[statId] = MAX_STAT_STAGE;
+        
+    if ((flags & STAT_CHANGE_PRESERVE_MULTICHOOSE) == 0)
```
Comment (DizzyEggg) [inline src/battle_script_commands.c:10628]: if (!(flags & STAT_CHANGE_PRESERVE_MULTICHOOSE))

**Review by DizzyEggg — CHANGES_REQUESTED** [review]
Fix conflicts

**Comment by Bassoonian** [comment]
@tustin2121 do you think you could update this PR? Thanks!

**Comment by Bassoonian** [comment]
Closed over inactivity. Feel free to open a new PR later.

**Pull request closed without merging**